src/Payum/Core/Bridge/Guzzle/HttpClientFactory.php

```diff
@@ -1,26 +1,42 @@
 <?php
+
 namespace Payum\Core\Bridge\Guzzle;
 
-use GuzzleHttp\Client as GuzzleClient;
-use GuzzleHttp\ClientInterface as GuzzleClientInterface;
+use Http\Discovery\HttpClientDiscovery;
 use Payum\Core\HttpClientInterface;
 
+/**
+ * @deprecated This will be removed in 2.0. Consider using Http\Discovery\HttpClientDiscovery.
+ */
 class HttpClientFactory
 {
     /**
-     * @return GuzzleClientInterface
+     * Create a Guzzle client.
+     *
+     * @return \GuzzleHttp\Client
      */
     public static function createGuzzle()
     {
+        trigger_error('The function "HttpClientFactory::createGuzzle" is depcrecated and will be removed in 2.0.', E_USER_DEPRECATED);
+
+        $client = null;
+        if (!class_exists('GuzzleHttp\Client')) {
```

Comment:
I can't. Guzzle might not be installed. If I'm using `Client::class` i would get an error saying "Class not found" instead of my custom exception.